Fix for AttributeError with libcloud <0.15

brejoc opened the pull request:
### What does this PR do?

Fixes an issue with libcloud <0.15 and CloudStack. The CloudStack implementation is using a method that was introduced with libcloud 0.15. But e.g. Ubuntu 12.04 is shipping with libcloud version 0.14.1.

An alternative would be pinning the version of libcloud to >=0.15 and maybe packaging and shipping libcloud in a version needed.
### What issues does this PR fix or reference?
#31729
### Previous Behavior

Crashed with AttributeError during VM creation. Noticed this with Exoscale as the provider.
### New Behavior

VM creation is now working.
### Tests written?
- [ ] Yes
- [x] No

brejoc committed:
Fix for AttributeError with libcloud <0.15

ex_list_projects is introduced in version 0.15 of libcloud. I'm simply
cathing the AttributeError here to return False, just like there would
be not project.

Fixes #31729

jfindlay commented on salt/cloud/clouds/cloudstack.py:
```diff
@@ -202,7 +202,11 @@ def get_project(conn, vm_):
+    try:
+        projects = conn.ex_list_projects()
+    except AttributeError:
+        # with versions <0.15 of libcloud this is causing an AttributeError.
```
@brejoc, can you also add something like this here: `log.warning('Cannot get projects, you may need to update libcloud to 0.15 or later')`?

brejoc replied:
@jfindlay sure!

brejoc committed:
Added log message with update suggestion for libcloud

Log message notifies user of old libcloud version and inability to get
projects.

salt-jenkins commented:
Can one of the admins verify this patch?

rallytime commented:
@brejoc Thank you for this change! That will be much cleaner to catch that error. Can you also update the `depends` statement at the top of the file to say `:depends: libcloud >= 0.15`? That is a small thing, but will be helpful in the long run. :)

brejoc committed:
Added version to libcloud depends statement

Now states ">= 0.15" instead of no version.

brejoc commented:
@rallytime No problem. I'm glad if this helps in the future!

rallytime commented:
Go Go Jenkins!
